dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java

```diff
@@ -170,4 +173,31 @@ public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscover
         AnnotatedTypeWrapper<T> wrapper = new AnnotatedTypeWrapper<T>(beanManager, processAnnotatedType.getAnnotatedType(), interceptedClass, interceptedMethods);
         processAnnotatedType.setAnnotatedType(wrapper);
     }
+
+    // Move the FaultToleranceInterceptor up before the Transactional interceptors if required
+    public void afterTypeDiscovery(@Observes AfterTypeDiscovery atd) {
+        if (Boolean.getBoolean("com.ibm.ws.faulttolerance.run.before.transactional")) {
+            int fti = -1;
+            int txi = -1;
+            int interceptor = 0;
+            for (Iterator<Class<?>> iterator = atd.getInterceptors().iterator(); iterator.hasNext() && !(fti >= 0 && txi >= 0);) {
+                Class<?> i = iterator.next();
+                if (FaultToleranceInterceptor.class.equals(i)) {
+                    fti = interceptor;
+                } else if (txi < 0) {
+                    if (TransactionalInterceptor.class.isAssignableFrom(i)) {
+                        txi = interceptor;
+                    }
+                }
+                interceptor++;
+            }
+
+            if (fti >= 0 && txi >= 0) {
```

Comment:
Since, you're changing this method anyway, please add a few more comments and slightly more descriptive variable names. It's not that easy to follow.

Reply:
Method is now more readable hopefully.